packages/cli/src/scaling/processor.ts

```diff
@@ -0,0 +1,188 @@
+				where: { id: workflowId },
+			});
+
+			if (workflowData === null) {
+				this.logger.error('[Processor] Failed to find workflow', { workflowId, executionId });
+				throw new ApplicationError('Failed to find workflow', { extra: { workflowId } });
+			}
+
+			staticData = workflowData.staticData;
+		}
+
+		const workflowSettings = execution.workflowData.settings ?? {};
+
+		let workflowTimeout = workflowSettings.executionTimeout ?? config.getEnv('executions.timeout');
+
+		let executionTimeoutTimestamp: number | undefined;
+
+		if (workflowTimeout > 0) {
+			workflowTimeout = Math.min(workflowTimeout, config.getEnv('executions.maxTimeout'));
+			executionTimeoutTimestamp = Date.now() + workflowTimeout * 1000;
+		}
+
+		const workflow = new Workflow({
+			id: workflowId,
+			name: execution.workflowData.name,
+			nodes: execution.workflowData.nodes,
+			connections: execution.workflowData.connections,
+			active: execution.workflowData.active,
+			nodeTypes: this.nodeTypes,
+			staticData,
+			settings: execution.workflowData.settings,
+		});
+
+		const additionalData = await WorkflowExecuteAdditionalData.getBase(
+			undefined,
+			undefined,
+			executionTimeoutTimestamp,
+		);
+
+		additionalData.hooks = WorkflowExecuteAdditionalData.getWorkflowHooksWorkerExecuter(
+			execution.mode,
+			job.data.executionId,
+			execution.workflowData,
+			{ retryOf: execution.retryOf as string },
+		);
+
+		additionalData.hooks.hookFunctions.sendResponse = [
+			async (response: IExecuteResponsePromiseData): Promise<void> => {
+				const webhookResponseReport: WebhookResponseReport = {
+					kind: 'webhook-response',
+					executionId,
+					response: this.encodeWebhookResponse(response),
+				};
+
+				await job.progress(webhookResponseReport);
+			},
+		];
+
+		additionalData.executionId = executionId;
+
+		additionalData.setExecutionStatus = (status: ExecutionStatus) => {
+			// Can't set the status directly in the queued worker, but it will happen in InternalHook.onWorkflowPostExecute
+			this.logger.debug(
+				`[Processor] Queued worker execution status for ${executionId} is "${status}"`,
+			);
+		};
+
+		let workflowExecute: WorkflowExecute;
+		let workflowRun: PCancelable<IRun>;
+		if (execution.data !== undefined) {
+			workflowExecute = new WorkflowExecute(additionalData, execution.mode, execution.data);
+			workflowRun = workflowExecute.processRunExecutionData(workflow);
+		} else {
+			// Execute all nodes
+			// Can execute without webhook so go on
+			workflowExecute = new WorkflowExecute(additionalData, execution.mode);
+			workflowRun = workflowExecute.run(workflow);
+		}
+
+		const runningJob: RunningJob = {
+			run: workflowRun,
+			executionId,
+			workflowId: execution.workflowId,
+			workflowName: execution.workflowData.name,
+			mode: execution.mode,
+			startedAt: execution.startedAt,
+			retryOf: execution.retryOf ?? '',
+			status: execution.status,
+		};
+
+		this.runningJobs[job.id] = runningJob;
+
+		await workflowRun;
```

Comment:
For now this PR is consolidating logic, but otherwise leaving it untouched. Let's interrogate this legacy code as we later refactor and write tests for it.